Lib/pathlib/_abc.py

```diff
@@ -846,6 +846,11 @@ def copytree(self, target, *, follow_symlinks=True,
         if on_error is None:
             def on_error(err):
                 raise err
+        if target.is_relative_to(self):
+            try:
+                raise OSError(f"Cannot copy {self!r} inside itself: {target!r}")
```

Comment:
I didn't check, but should you set filename and filename2 for this? (it could be useful for callbacks).